tests/test_history_dummy.py

```diff
@@ -10,3 +10,19 @@
 def test_construct_history_str(xession, backend):
     xession.env["XONSH_HISTORY_BACKEND"] = backend
     assert isinstance(construct_history(), DummyHistory)
+
+
+@pytest.mark.parametrize("backend", ["dummy", DummyHistory, DummyHistory()])
+def test_ignore_regex_invalid(xession, backend):
+    xession.env["XONSH_HISTORY_BACKEND"] = backend
+    xession.env["XONSH_IGNORE_REGEX"] = "**"
```

Comment:
I can't figure out how to inject `XONSH_IGNORE_REGEX` -- I don't think that `XSH.env` in the `History` class can see it.

Reply:
setting them here is fine. It should be reflected during test run. You can see similar usage by other tests

Reply:
In this case empirically it did not work. I'll hunt around and also make sure I didn't spell something wrong.

Reply:
`XONSH_HISTORY_IGNORE_REGEX` != `XONSH_IGNORE_REGEX` 🤦